spaceapi/src/lib.rs

```diff
@@ -171,42 +195,18 @@ impl ToJson for Status {
         d.insert("location".to_string(), self.location.to_json());
         d.insert("contact".to_string(), self.contact.to_json());
 
-        match self.spacefed {
-            Optional::Value(ref spacefed) => { d.insert("spacefed".to_string(), spacefed.to_json()); },
-            Optional::Absent => (),
-        };
-        match self.projects {
-            Optional::Value(ref projects) => { d.insert("projects".to_string(), projects.to_json()); },
-            Optional::Absent => (),
-        };
-        match self.cam {
-            Optional::Value(ref cam) => { d.insert("cam".to_string(), cam.to_json()); },
-            Optional::Absent => (),
-        };
-        match self.feeds {
-            Optional::Value(ref feeds) => { d.insert("feeds".to_string(), feeds.to_json()); },
-            Optional::Absent => (),
-        };
-        match self.events {
-            Optional::Value(ref events) => { d.insert("events".to_string(), events.to_json()); },
-            Optional::Absent => (),
-        };
-        match self.radio_show {
-            Optional::Value(ref radio_show) => { d.insert("radio_show".to_string(), radio_show.to_json()); },
-            Optional::Absent => (),
-        };
+        self.spacefed.as_ref().map_or((), |v| { d.insert("spacefed".to_string(), v.to_json()); } );
+        self.projects.as_ref().map_or((), |v| { d.insert("projects".to_string(), v.to_json()); });
+        self.cam.as_ref().map_or((), |v| { d.insert("cam".to_string(), v.to_json()); });
+        self.feeds.as_ref().map_or((), |v| { d.insert("feeds".to_string(), v.to_json()); });
+        self.events.as_ref().map_or((), |v| { d.insert("events".to_string(), v.to_json()); });
+        self.radio_show.as_ref().map_or((), |v| { d.insert("radio_show".to_string(), v.to_json()); });
```

Comment:
I fully agree, but at least it reduces the 4 line ``match`` statement to a single line.
The target would of course be something like ``#[derive(ToJson)]``